### ISSUE TITLE
Is 'doc++' still in use for Stratagus?

### ISSUE THREAD

**Issue by timfel** [Wargus/stratagus#20]
File "doc/development.html" says that "doc++" can be used, 
but when checking files from package "stratagus_2.2.5.5.orig.tar.gz"
I've found that 

1) makefile isn't able to do handle  "make doc++" as described;

2) link 'DOC++' inside file "doc/development.html"  is broken :(
Should it point to "http://docpp.sourceforge.net/" instead? (or even be removed?)

The following files refer to doc++:
~/src/stratagus/stratagus.cpp
~/doc/development.html

It's just a small documentation issue with doc++, 
the doxygen stuff is being generetade properly :)

Best regards, 
Tilpa

------------------------------------
Imported from Launchpad using lp2gh.

 * date created: 2011-03-03T22:00:20Z
 * owner: fabio-pasini
 * assignee: pali
 * the launchpad url was https://bugs.launchpad.net/bugs/728715

**Comment by timfel** [comment]
(by fabio-pasini)
Doc++ stuff got removed from makefile at
http://bazaar.launchpad.net/~stratagus/stratagus/trunk/revision/6744

Still hadn't time to check if that code still works with the latest doc++ version, 
I can take a look next thursday - it's Carnival here in Brazil, and I don't have access to my ubuntu machine 'till end of the holydays.
[]'s
Tilpa

**Comment by timfel** [comment]
(by fabio-pasini)
The makefile chunk related to doc++ as of revision 6744 doesn't work anymore :(
I'll make some more tests  and post a patch here  if I got something useful. 

Folks, you think this issue doesn't worth a fix, just let me know.
[]'s
Tilpa

**Comment by timfel** [comment]
(by pali)
Our static Makefile(s) will be replaced by CMake in future. So all changes in Makefile(s) will be deleted! If you want support doc++ please add rules to new CMake scripts.

Bzr branch for CMake build system: https://code.launchpad.net/~stratagus/stratagus/stratagus-cmake
When it be stable it will be merged into trunk.

**Comment by timfel** [comment]
(by fabio-pasini)
Starting doc++ as described in the attached file  
("generate_doc++.sh", run it from the same directory as makefile ) 
it spends a really long time eating 99% cpu before output the results.

I tried changing the input file order sequence etc, with the same results.

There are also other (unrelated) bugs 
for current doc++ stable version (3.4.10, released on December 22, 2002), 
e.g. http://www.lorene.obspm.fr/oldlist/2003-November/000012.html

My vote is to simply remove all doc++ references from files
~/src/stratagus/stratagus.cpp 
~/doc/development.html,
and keep not using doc++.

In fact, I think that doxygen documentation looks way better than doc++ output.

**Comment by timfel** [comment]
(by pali)
I think that doc++ nobody used. If doxygen works fine, doc++ could be deleted.

**Comment by timfel** [comment]
(by pali)
Stratagus now using CMake build system. Doc++ support was dropped, CMake script support generating Doxygen documentation.

**Comment by timfel** [comment]
(by fabio-pasini)
>Make script support generating Doxygen documentation.
cmake is generating doxygen documantation properly 
(as of revision 7953, great work, dudes!), 
just a few more adjustments needed to reflect the new build mecanism:

# Broken links:
1) Files "/doc/index.html"  states
"After Stratagus compilation can be found in the source code documentation", 
but html link "source code documentation" points to the wrong place.
2) Also, file "/doc/development.html"  states
"After compilation is availabe here"
but html link "here" points to the wrong place.
Question: 
Should those links be corrected and point to "/build/doxygen/index.html", 
or should cmake generate doxygen files into "doc/doxygen/" ?
(I do prefer the latter, but its up to you, guys :)

# Document new usage:
3) Doxygen files are generated only after set the option "ENABLE_DOC" to "ON" 
inside file "CMakeLists.txt", 
but that isn't mentioned anywere. 
I believe it would be nice to enhance "/doc/development.html" and
some more details about where and how change the build scripts to have 
doxygen stuff being created.

**Comment by timfel** [comment]
(by pali)
1) and 2)
you need to run make install. it copy html files from doc and from buildir/doxygen. problem is that you can create build directory not only in <root_folder>/build and is not good idea to generate some files outside build directory...

3)
I updated some info... but not all. and you can write detailed info too :)

**Comment by timfel** [comment]
(by fabio-pasini)
>and you can write detailed info too :)
You're right! :)
Please refer to file doc-proposed.tar.gz attached, I've included some more details about where find doxygen output, how generate it, and where configure it.
Those files are just a suggestion, feel free to change/adjust anything.
[]'s

**Comment by timfel** [comment]
(by pali)
Note that editing Doxyfine.in is not good idea. It is template for CMake (CMake generate correct Doxyfile from Doxyfile.in)
And as I wrote you need to run make install (with ENABLE_DOC) for correct paths.

make install copy <stratagus>/doc and <buildir>/doxygen to one install folder <prefix>/share/doc/stratagus/

**Comment by timfel** [comment]
(by fabio-pasini)
> Note that editing Doxyfine.in is not good idea.
OK, removed. 
.
> need to run make install (with ENABLE_DOC) for correct paths.
You're right, but ... 'make install'  is not  referred anywhere :(
I believe 'make install'  should be pointed inside install.htm, 
but I'm not experienced enough  to create docs for this part.
.
Added another proposal as attachment. Again, feel free to change/edit it as needed :)
[]'s

**Comment by timfel** [comment]
(by pali)
Ok. Now I updated info in html files. See commit: http://bazaar.launchpad.net/~stratagus/stratagus/trunk/revision/7990
